skimage/future/graph/rag.py

```diff
@@ -51,21 +41,79 @@ def min_weight(graph, src, dst, n):
     return min(w1, w2)
 
 
+def _add_edge_filter(values, graph):
+    """Create edge in `g` between the first element of `values` and the rest.
+
+    Add an edge between the first element in `values` and
+    all other elements of `values` in the graph `g`. `values[0]`
+    is expected to be the central value of the footprint used.
+
+    Parameters
+    ----------
+    values : array
+        The array to process.
+    graph : RAG
+        The graph to add edges in.
+
+    Returns
+    -------
+    0 : float
+        Always returns 0. The return value is required so that `generic_filter`
+        can put it in the output array, but it is ignored by this filter.
+    """
+    values = values.astype(int)
+    center = values[len(values) // 2]
+    for value in values:
+        if value != center and not graph.has_edge(center, value):
+            graph.add_edge(center, value)
+    return 0.
+
+
 class RAG(nx.Graph):
 
     """
     The Region Adjacency Graph (RAG) of an image, subclasses
     `networx.Graph <http://networkx.github.io/documentation/latest/reference/classes.graph.html>`_
+
+    Parameters
+    ----------
+    label_image : array of int
+        An initial segmentation, with each region labeled as a different
+        integer. Every unique value in ``label_image`` will correspond to
+        a node in the graph.
+    connectivity : int in {1, ..., ``label_image.ndim``}, optional
+        The connectivity between pixels in ``label_image``. For a 2D image,
+        a connectivity of 1 corresponds to immediate neighbors up, down,
+        left, and right, while a connectivity of 2 also includes diagonal
+        neighbors. See `scipy.ndimage.generate_binary_structure`.
+    data : networkx Graph specification, optional
+        Initial or additional edges to pass to the NetworkX Graph
+        constructor. See `networkx.Graph`. Valid edge specifications
+        include edge list (list of tuples), NumPy arrays, and SciPy
+        sparse matrices.
+    **attr : keyword arguments, optional
+        Additional attributes to add to the graph.
     """
 
-    def __init__(self, data=None, **attr):
+    def __init__(self, label_image=None, connectivity=1, data=None, **attr):
 
         super(RAG, self).__init__(data, **attr)
-        try:
-            self.max_id = max(self.nodes_iter())
-        except ValueError:
-            # Empty sequence
+        if self.number_of_nodes() == 0:
             self.max_id = 0
+        else:
+            self.max_id = max(self.nodes_iter())
+
+        if label_image is not None:
+            fp = ndi.generate_binary_structure(label_image.ndim, connectivity)
+            ndi.generic_filter(
+                label_image,
+                function=_add_edge_filter,
+                footprint=fp,
+                mode='nearest',
+                output=as_strided(np.empty((1,), dtype=np.float_),
```

Comment:
I find it hard to understand as well. Is this output needed because ``_add_edge_filter`` returns 0?